README.md

```diff
@@ -1,41 +1,150 @@
+| `page`      | The highest order context in the app. Describes the page / tab, such as Swap, Explore, Pool, Vote, etc.                               |
+| `section`   | A section within the top level context. When a modal is open, section describes the part of the page from which the modal originated. |
+| `modal`     | A modal, if one exists at the time of the event.                                                                                      |
+| `element`   | The most specific element that triggered the event such as a specific button or link.                                                 |
+
+No specific context is required and any or all of these trace fields may be null. Use these as it makes sense for your context, and define possible values in this repository. These fields will work in tandem with your event names and event properties.
+
+## Naming Conventions
+
+Event names should:
+- follow the Object-Action naming convention (based on this [article](https://segment.com/academy/collecting-data/naming-conventions-for-clean-data/)).
+- be human readable
+- capitalize each word (e.g [Proper Case](https://www.computerhope.com/jargon/p/proper-case.htm))
+- start with object/product to ensure events are grouped together in Amplitude (sorted alphabetically)
+- us past tense verbs (e.g. Button *Clicked*).
+- *not* use acronyms or short hand (e.g. use `Transaction` over `Txn`)
 
-## Documentation
+Property names should:
+- be `snake_case`
+- be prefixed with `is_` when representing a boolean value
+- end in units (unabbreviated) when needed (e.g. `time_seconds`)
+- *not* use acronyms or short hand (e.g. use `Transaction` over `Txn`)
 
-The top level folders such as [Interface](./src/interface) contain application specific data. This data gives context to the event being logged. Add a new folder for every new application being added.
+Property values should:
+- be a boolean, number, string, or enum (of strings) type.
+- strings: be consistent within all possible values, opting for either `Proper Case`, `kebab-case`,  or `snake-case`. The default selection should be `Proper Case`.
 
-The top level files, such as [primitives](./src//primitives.ts) contain generic data that is supposed to be reused by different applications. This data describes the events being logged, such as `EventName.MENU_CLICK`, which reflects an event on a menu. Specific event context should be captured in additional fields beyond the event name to create user-friendly hierarchy groupings.
+## Defining Events In Code
 
-## Example Apps
+Event names, property names, and property values should all be defined in Javascript using `enum`'s and follow the following conventions:
+- `enum` names should be as specific as possible (e.g `DocsProtocolVersion` over `Version`).
+- `enum` names should be in `PascalCase`.
+- Code values of enums should be named in `UPPER_SNAKE_CASE`.
```

Comment:
These are enforced by code, so no need to add them here - they add more complication.

Reply:
I think it's good to have this explicit even if also enforced in code!